src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/ApplicationDeployerFactory.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Server.IntegrationTesting.IIS;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Server.IntegrationTesting
+{
+    /// <summary>
+    /// Factory to create an appropriate deployer based on <see cref="DeploymentParameters"/>.
+    /// </summary>
+    public class IISApplicationDeployerFactory
+    {
+        /// <summary>
+        /// Creates a deployer instance based on settings in <see cref="DeploymentParameters"/>.
+        /// </summary>
+        /// <param name="deploymentParameters"></param>
+        /// <param name="loggerFactory"></param>
+        /// <returns></returns>
+        public static ApplicationDeployer Create(DeploymentParameters deploymentParameters, ILoggerFactory loggerFactory)
```

Comment:
The name IISApplicationDeployerFactory bugs me but I don't have a better name. What if you made it an extension method on DeploymentParameters instead? Then you wouldn't see the name, and it'd be shorter.
`deploymentParameters.CreateServer(loggerFactory)`